### PR TITLE
🎉 Support connecting with websocket proxies

### PR DESCRIPTION
**Pull request by awfulcooking**
So we can run in browsers (js/wasm)

Reading #11, you're gonna love this! :-)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,7 @@ module github.com/makeworld-the-better-one/go-gemini
 go 1.14
 
 require (
-	github.com/google/go-cmp v0.3.1
+	github.com/google/go-cmp v0.4.0
```
Comment (awfulcooking) [inline go.mod:6]: I didn't ask Go to do this.

Comment (awfulcooking) [reply go.mod:6]: Oh, it's a minor version, not major.

**Comment by makeworld-the-better-one** [comment]
Thanks for your PR. A websocket proxy is an obscure use case, and adds unneeded additional dependencies. IMO the ideal way to do this would be to add a configuration option, where you can set a variable in the configuration struct. That variable would be of the type `func() net.Conn, error`. If it's not `nil`, the code would run that function and use the returned socket instead of the default one.

The websocket specific code would go in your own repo. How does that sound?

**Comment by awfulcooking** [comment]
That sounds like a great solution, yes

**Comment by awfulcooking** [comment]
I'll rework this PR into that.

**Comment by makeworld-the-better-one** [comment]
Thanks!

**Pull request closed without merging**